scene/animation/animation_player.cpp

```diff
@@ -641,6 +695,47 @@ double AnimationPlayer::get_current_animation_length() const {
 	return playback.current.from->animation->get_length();
 }
 
+void AnimationPlayer::set_current_section_with_markers(const StringName &p_start_marker, const StringName &p_end_marker) {
+	ERR_FAIL_NULL_MSG(playback.current.from, "AnimationPlayer has no current animation.");
+	ERR_FAIL_COND_MSG(p_start_marker == p_end_marker && p_start_marker, vformat("Start marker and end marker cannot be the same marker: %s.", p_start_marker));
+	ERR_FAIL_COND_MSG(p_start_marker && !playback.current.from->animation->has_marker(p_start_marker), vformat("Marker %s not found in animation: %s.", p_start_marker, playback.current.from->animation->get_name()));
+	ERR_FAIL_COND_MSG(p_end_marker && !playback.current.from->animation->has_marker(p_end_marker), vformat("Marker %s not found in animation: %s.", p_end_marker, playback.current.from->animation->get_name()));
+	double start_time = p_start_marker ? playback.current.from->animation->get_marker_time(p_start_marker) : -1;
+	double end_time = p_end_marker ? playback.current.from->animation->get_marker_time(p_end_marker) : -1;
+	set_current_section(start_time, end_time);
+}
+
+void AnimationPlayer::set_current_section(double p_start_time, double p_end_time) {
+	ERR_FAIL_NULL_MSG(playback.current.from, "AnimationPlayer has no current animation.");
+	ERR_FAIL_COND_MSG(p_start_time >= 0 && p_end_time >= 0 && Animation::is_greater_approx(p_start_time, p_end_time), vformat("Start time %f is greater than end time %f.", p_start_time, p_end_time));
+	playback.current.start_time = p_start_time;
+	playback.current.end_time = p_end_time;
+	playback.current.pos = CLAMP(playback.current.pos, get_current_section_start_time(), get_current_section_end_time());
+}
+
+void AnimationPlayer::reset_current_section() {
+	playback.current.start_time = -1;
+	playback.current.end_time = -1;
+}
+
+double AnimationPlayer::get_current_section_start_time() const {
+	if (playback.current.start_time < 0) {
+		return 0;
+	}
+	return playback.current.start_time;
+}
+
+double AnimationPlayer::get_current_section_end_time() const {
+	if (playback.current.end_time < 0) {
+		return playback.current.from->animation->get_length();
+	}
+	return playback.current.end_time;
+}
+
+bool AnimationPlayer::is_playing_section() const {
+	return playback.current.start_time != -1 || playback.current.end_time != -1;
+}
+
```

Comment:
```suggestion
void AnimationPlayer::set_current_section_with_markers(const StringName &p_start_marker, const StringName &p_end_marker) {
	ERR_FAIL_NULL_MSG(playback.current.from, "AnimationPlayer has no current animation.");
	ERR_FAIL_COND_MSG(p_start_marker == p_end_marker && p_start_marker, vformat("Start marker and end marker cannot be the same marker: %s.", p_start_marker));
	ERR_FAIL_COND_MSG(p_start_marker && !playback.current.from->animation->has_marker(p_start_marker), vformat("Marker %s not found in animation: %s.", p_start_marker, playback.current.from->animation->get_name()));
	ERR_FAIL_COND_MSG(p_end_marker && !playback.current.from->animation->has_marker(p_end_marker), vformat("Marker %s not found in animation: %s.", p_end_marker, playback.current.from->animation->get_name()));
	double start_time = p_start_marker ? playback.current.from->animation->get_marker_time(p_start_marker) : -1;
	double end_time = p_end_marker ? playback.current.from->animation->get_marker_time(p_end_marker) : -1;
	ERR_FAIL_COND_MSG(p_start_marker && Animation::is_less_approx(start_time, 0), vformat("Marker %s time must be positive in animation: %s.", p_start_marker, playback.current.from->animation->get_name()));
	ERR_FAIL_COND_MSG(p_end_marker && Animation::is_less_approx(end_time, 0), vformat("Marker %s time must be positive in animation: %s.", p_end_marker, playback.current.from->animation->get_name()));
	set_current_section(start_time, end_time);
}

void AnimationPlayer::set_current_section(double p_start_time, double p_end_time) {
	ERR_FAIL_NULL_MSG(playback.current.from, "AnimationPlayer has no current animation.");
	ERR_FAIL_COND_MSG(Animation::is_greater_or_equal_approx(p_start_time, 0) && Animation::is_greater_or_equal_approx(p_end_time, 0) && Animation::is_greater_approx(p_start_time, p_end_time), vformat("Start time %f is greater than end time %f.", p_start_time, p_end_time));
	ERR_FAIL_COND_MSG(Animation::is_greater_approx(p_start_time, length), vformat("Start time %f must be lesser than the length of animation: %s.", p_start_time, playback.current.from->animation->get_name()));
	playback.current.start_time = p_start_time;
	playback.current.end_time = p_end_time;
	playback.current.pos = CLAMP(playback.current.pos, get_current_section_start_time(), get_current_section_end_time());
}

void AnimationPlayer::reset_current_section() {
	playback.current.start_time = -1;
	playback.current.end_time = -1;
}

double AnimationPlayer::get_current_section_start_time() const {
	if (Animation::is_less_approx(playback.current.start_time, 0)) {
		return 0;
	}
	return playback.current.start_time;
}

double AnimationPlayer::get_current_section_end_time() const {
	if (Animation::is_less_approx(playback.current.end_time, 0)) {
		return playback.current.from->animation->get_length();
	}
	return playback.current.end_time;
}

bool AnimationPlayer::is_playing_section() const {
	return Animation::is_greater_or_equal_approx(playback.current.start_time, 0) || Animation::is_greater_or_equal_approx(playback.current.end_time, 0);
}

```
It is better to have an error if there is a possibility of setting an invalid time.